numba/core/typeconv/rules.py

```diff
@@ -30,9 +30,11 @@ def _init_casting_rules(tm):
     tcr.safe_unsafe(types.uint16, types.int32)
     tcr.safe_unsafe(types.uint32, types.int64)
 
+    tcr.safe_unsafe(types.int8, types.float16)
```

Comment:
Is conversion from `int16` to `float16` also possible? I note from the limits of a `float16` and `int16` that the range of values of an `int16` seems to fit into the range of a `float16`:

```
print(np.finfo(np.float16))
# Machine parameters for float16
# ---------------------------------------------------------------
# precision =   3   resolution = 1.00040e-03
# machep =    -10   eps =        9.76562e-04
# negep =     -11   epsneg =     4.88281e-04
# minexp =    -14   tiny =       6.10352e-05
# maxexp =     16   max =        6.55040e+04
# nexp =        5   min =        -max
# ---------------------------------------------------------------

print(np.iinfo(np.int16))
# Machine parameters for int16
# ---------------------------------------------------------------
# min = -32768
# max = 32767
# ---------------------------------------------------------------
```

Reply:
Actually nevermind, I see that that conversion is already registered on line L36/R38.

Reply:
@gmarkall I modeled int16 to float16 after int32 to float32 which is considered unsafe:

tcr.unsafe_unsafe(types.int32, types.float32)